core/OpenStore365.Serverless.Auth/AAD_B2C/HttpHeaderUserProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.AspNetCore.Http;
+using OpenStore365.Serverless.Auth.AAD_B2C.Constants;
+using System;
+
+namespace OpenStore365.Serverless.Auth.AAD_B2C
+{
+    public class HttpHeaderUserProvider : IUserProvider
+    {
+        public User GetUser(HttpRequest httpRequest) =>
+            new User(
+                ExtractUserId(httpRequest),
+                TryExtractDisplayName(httpRequest));
+
+        private string ExtractUserId(HttpRequest httpRequest)
+        {
+            if (!httpRequest.Headers.TryGetValue(EasyAuth.HttpHeaders.ID, out var id))
+            {
+                throw new Exception($"User ID must be present in the '{EasyAuth.HttpHeaders.ID}' HTTP header.");
+            }
+
+            if (!Guid.TryParse(id, out var _))
+            {
+                throw new Exception("User ID has to follow the GUID format.");
+            }
```

Comment:
Wouldn't it be better to have a defined class that derives from Exception? Such as InvalidGuidException etc.?

Reply:
It would be beneficial in case of more advanced logging mechanism or global exception handling. Logging is not the main focus of this particular sample. Global exception handling is not supported by Azure Functions out of the box due to it not having a middleware pipeline.